scheduler/test/cook/test/rest/api.clj

```diff
@@ -1392,6 +1393,26 @@
             (is (= (expected-job-map job)
                    (dissoc (api/fetch-job-map (db conn) uuid) :submit_time)))))
 
+        (testing "should work when the job specifies checkpointing options"
+          (let [conn (restore-fresh-database! "datomic:mem://mesos-api-test")
+                checkpoint {:mode "auto"}
+                {:keys [uuid] :as job} (assoc (minimal-job) :checkpoint checkpoint)]
+            (is (= {::api/results (str "submitted jobs " uuid)}
+                   (testutil/create-jobs! conn {::api/jobs [job]})))
+            (is (= (expected-job-map job)
+                   (dissoc (api/fetch-job-map (db conn) uuid) :submit_time)))))
+
+        (testing "should work when the job specifies checkpointing options 2"
+          (let [conn (restore-fresh-database! "datomic:mem://mesos-api-test")
+                checkpoint {:mode "periodic"
+                            :options {:preserve-paths #{"p1" "p2"}}
+                            :periodic-options {:period-sec 777}}
+                {:keys [uuid] :as job} (assoc (minimal-job) :checkpoint checkpoint)]
+            (is (= {::api/results (str "submitted jobs " uuid)}
+                   (testutil/create-jobs! conn {::api/jobs [job]})))
+            (is (= (expected-job-map job)
+                   (dissoc (api/fetch-job-map (db conn) uuid) :submit_time)))))
+
```

Comment:
Should there be a test for `preemptible` mode?

Reply:
I don't think so. No special options for it. Just a string